### DISCUSSION TITLE
Fix datapacks

### DISCUSSION THREAD

**Discussion by Nicolol1234** [PurpurMC/Purpur#1314] [__Dismissed__] [answered]
Adding an option or fix the server software for datapacks to work again, because the Paper API break some datapacks, and it's __realy__ anoying to choose between plugins or datapacks

**Comment by RhythmicSys** [comment, marked as answer]
This is unfortunately not possible. 
On bukkit/spigot/paper etc, the types of data packs that will work are things that are data driven: world generation, recipes, advancements, loot tables, tags

But anything that uses functions requires vanilla/fabric because bukkit and forks have make a lot of optimizations for multiplayer play, which is what breaks the data packs. 

And because we build on top of these forks as our foundation we can't really undo or fix that
